x-pack/plugins/siem/server/lib/kpi_hosts/query_authentication.dsl.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { createQueryFilterClauses } from '../../utils/build_query';
+import { RequestBasicOptions } from '../framework';
+
+import { KpiHostsESMSearchBody } from './types';
+
+const getAuthQueryFilter = () => [
+  {
+    bool: {
+      should: [
+        {
+          match: {
+            'event.type': 'authentication_success',
+          },
+        },
+        {
+          match: {
+            'event.type': 'authentication_failure',
+          },
+        },
+      ],
+      minimum_should_match: 1,
+    },
+  },
+];
+
+export const buildAuthQuery = ({
+  filterQuery,
+  timerange: { from, to },
+  sourceConfiguration: {
+    fields: { timestamp },
+    logAlias,
+    auditbeatAlias,
+    packetbeatAlias,
+    winlogbeatAlias,
+  },
+}: RequestBasicOptions): KpiHostsESMSearchBody[] => {
+  const filter = [
+    ...createQueryFilterClauses(filterQuery),
+    ...getAuthQueryFilter(),
+    {
+      range: {
+        [timestamp]: {
+          gte: from,
+          lte: to,
+        },
+      },
+    },
+  ];
+
+  const dslQuery = [
+    {
+      index: [logAlias, auditbeatAlias, packetbeatAlias, winlogbeatAlias],
+      allowNoIndices: true,
+      ignoreUnavailable: true,
+    },
+    {
+      aggs: {
+        authentication_success: {
+          filter: {
+            term: {
+              'event.type': 'authentication_success',
+            },
+          },
+        },
+        authentication_success_histogram: {
+          auto_date_histogram: {
```

Comment:
Sometimes this is not giving me the bucket number I request for, is this the right way to ask for data?

Reply:
I'm not super familiar with the [Auto-Interval Date Histogram Agg](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-autodatehistogram-aggregation.html), but the docs note that:

>"the number of buckets returned will always be less than or equal to this target number.

If you want specific intervals you could use the regular [Date Histogram Agg](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-bucket-datehistogram-aggregation.html). As for the _right way_ to ask for this data, I think it really depends on what we want to show here. What you have seems 👍 though!

Reply:
Lovely, as I am not sure I want to handle the interval by myself here, so letting api does the job.